props-aws/src/main/java/sh/props/aws/AwsSecretsManager.java

```diff
@@ -223,24 +222,11 @@ void scheduleSecretsRetrieval(List<String> allSecretIds) {
       futures.add(future);
     }
 
-    // for now, wait for all secrets to be retrieved before returning
-    // TODO: need a smarter mechanism to handle partial updates of the source
-    for (var future : futures) {
-      try {
-        future.join();
-      } catch (Exception e) {
-        log.log(Level.WARNING, e, () -> "Unexpected exception while retrieving a secret");
-      }
-    }
-  }
-
-  /** Triggers a {@link #get()} call and sends all values to the registered downstream consumers. */
-  // TODO: look at implementing this method disconnected from this.get()
-  @Override
-  public void updateSubscribers() {
-    Map<String, String> data = Collections.unmodifiableMap(this.get());
-    for (Consumer<Map<String, String>> subscriber : this.subscribers) {
-      subscriber.accept(data);
+    // wait for all secrets to be retrieved before returning
+    try {
+      CompletableFuture.allOf(futures.toArray(CompletableFuture[]::new)).join();
```

Comment:
drive-by: use `allOf` to wait for all futures; I had crudely implemented it before as a for-loop